src/Symfony/Component/Validator/Constraints/ImageValidator.php

```diff
@@ -234,4 +247,46 @@ public function validate(mixed $value, Constraint $constraint)
             imagedestroy($resource);
         }
     }
+
+    /**
+     * Check whether a value is an SVG image
+     *
+     * @param mixed $value
+     *
+     * @return bool
+     * <b>TRUE</b> if <i>value</i> is an SVG, <b>FALSE</b> if it's not, or if we can't detect its MimeType;
+     */
+    private function isSvg(mixed $value)
+    {
+        if ($value instanceof FileObject) {
+            $mime = $value->getMimeType();
+        } elseif (class_exists(MimeTypes::class)) {
+            $mime = MimeTypes::getDefault()->guessMimeType($value);
+        } elseif (!class_exists(FileObject::class)) {
+            return false;
+        } else {
+            $mime = (new FileObject($value))->getMimeType();
+        }
+
+        return 'image/svg+xml' === $mime;
+    }
+
+    /**
+     * @param mixed $value
+     *
+     * @return array|null
+     * <b>array</b> contains the width and height of the image
+     * <b>null</b> if the XML is corrupted
+     */
+    private function getSvgSize(mixed $value)
+    {
+        if (false === $xmlValue = simplexml_load_file($value)) {
+            return null;
+        }
+        $svgAttributes = $xmlValue->attributes();
+        $width = intval($svgAttributes->width);
+        $height = intval($svgAttributes->height);
+
+        return [$width, $height];
```

Comment:
I don't know if it's common for others, but I usually work with SVG files and the following is not that uncommon:

* To not have any `width` or `height` attribute and just rely on the `viewBox` attribute dimensions
* To have values like `100%` in the `width` or `height` attributes because those images are embedded in other parent elements with some height/width defined

Reply:
I agree with @javiereguiluz here. Does it mean that this is not that useful? The size of an SVG is not that important anyway, maybe just the ratio? cc @symfony/mergers

Reply:
> maybe just the ratio?

I agree this one would be important.

Reply:
Just to be clear, I would only implement the ratio validation for SVGs.

Reply:
I would be fine to also validate width and height if we make sure to skip this validation in case the given width or height are relative values (like mentioned by Javier).

Reply:
But I fail to see how this is useful. A 100x200 SVG file looks perfectly fine at 20x40 or 2000x4000.